onnxruntime/core/platform/windows/telemetry.h

```diff
@@ -0,0 +1,57 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+#pragma once
+#include "core/platform/telemetry.h"
+#include "core/platform/ort_mutex.h"
+#include <atomic>
+
+// ***
+// platform specific control bits
+#ifndef TraceLoggingOptionMicrosoftTelemetry
+#define TraceLoggingOptionMicrosoftTelemetry() \
+  TraceLoggingOptionGroup(0000000000, 00000, 00000, 0000, 0000, 0000, 0000, 0000, 000, 0000, 0000)
+#endif
+#define MICROSOFT_KEYWORD_MEASURES       0x0000400000000000  // Bit 46
+#define TelemetryPrivacyDataTag(tag)     TraceLoggingUInt64((tag), "PartA_PrivTags")
+#define PDT_ProductAndServicePerformance 0x0000000001000000u
+#define PDT_ProductAndServiceUsage       0x0000000002000000u
+// ***
+
+namespace onnxruntime {
+
+/**
+  * derives and implments a Telemetry provider on Windows
+  */
+class WindowsTelemetry : public Telemetry {
+
+ public:
+
+  // these are allowed to be created, WindowsEnv will create one
+  WindowsTelemetry();
+  ~WindowsTelemetry();
+
+  void EnableTelemetryEvents() const override;
+  void DisableTelemetryEvents() const override;
+
+  void LogProcessInfo() const override;
+
+  void LogSessionCreation(uint32_t session_id, int64_t ir_version, const std::string& model_producer_name,
```

Comment:
Would it be better to use a struct to pass so many parameters to the function? This way if we add one more thing to the event only the struct needs to change, not the function declarations and definitions across several files.

Reply:
i like it.  want me to do that in this PR, or would as second change be ok ?  another thing to keep in mind is that changing this is a big deal.   it's a "schema change" .  so this is not somethin we do all the time, or need to necessarily optimize for.     in our experience we change it *maybe* once a year.   as an example winML is coming on 4 releases now (2 years) and we haven't change the schema yet.

---
In reply to: [335472938](https://github.com/microsoft/onnxruntime/pull/2109#discussion_r335472938) [](ancestors = 335472938)

Reply:
I think it's going to take us some iterations before we stabilize and it would be nice to avoid the pain of changing across multiple files in the meanwhile.

Reply:
i added it as a work item. thx

---
In reply to: [335778701](https://github.com/microsoft/onnxruntime/pull/2109#discussion_r335778701) [](ancestors = 335778701)